aten/src/ATen/native/cuda/UnarySignKernels.cu

```diff
@@ -51,9 +51,26 @@ void signbit_kernel_cuda(TensorIterator& iter){
   });
 }
 
+template<typename T>
+__host__ __device__ static inline c10::complex<T> sgn_wrapper(c10::complex<T> z) {
```

Comment:
this could be an alternate way of implementing this function:
```
template<typename T>
__host__ __device__ static inline c10::complex<T> sgn_wrapper(c10::complex<T> z) {
  if (z == c10::complex<T>(0, 0)) {
    return c10::complex<T>(0, 0);
  } else {
    auto angle = std::arg(z);
    auto cos = std::cos(angle);
    auto mod_sin = std::sqrt(1 - cos*cos);
    auto sin = (z.real() > 0) ? mod_sin : mod_sin * -1;
    return c10::complex<T>(cos, sin);
  }
}
template<typename T>
__host__ __device__ static inline c10::complex<T> sgn_wrapper(c10::complex<T> z) {
  if (z == c10::complex<T>(0, 0)) {
    return c10::complex<T>(0, 0);
  } else {
    auto angle = std::arg(z);
    auto cos = std::cos(angle);
    auto mod_sin = std::sqrt(1 - cos*cos);
    auto sin = (z.real() > 0) ? mod_sin : mod_sin * -1;
    return c10::complex<T>(cos, sin);
  }
}
``` 

The performance for the two versions are:
x/ |x| :  

```
In [2]: x=torch.randn(100, dtype=torch.cfloat)

In [3]: %timeit x.sgn()
60.5 µs ± 1.2 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [4]: %timeit x.cuda().sgn()
250 µs ± 11.6 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

above mentioned implementation:

```
In [2]: x=torch.randn(100, dtype=torch.cfloat)

In [5]:  %timeit x.cuda().sgn()
242 µs ± 4.25 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [6]: %timeit x.cuda().sgn()
237 µs ± 5.1 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)

In [7]: %timeit x.cuda().sgn()
243 µs ± 5.44 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

cc. @ngimel @ezyang

Reply:
1) Please use `Timer` utility for cuda measurements, %timeit is not doing necessary synchronization https://github.com/pytorch/pytorch/tree/master/torch/utils/_benchmark. 
2) benchmark on the variety of sizes, benchmarking on the tensor of size 100 is not particularly informative. 
3) you are benchmarking time to transfer tensor to the GPU + time of the operation, that's probably not what you want.